products/stratos/modules/integration/test-common/src/main/java/org/apache/stratos/integration/common/extensions/StratosServerExtension.java

```diff
@@ -107,11 +106,10 @@ private void startStratosServer(int activeMQDynamicPort) throws AutomationFramew
             long time3 = System.currentTimeMillis();
             String carbonHome = stratosTestServerManager.startServer();
             assertNotNull(carbonHome, "CARBON_HOME is null");
-            while (!serverStarted()) {
-                log.info("Waiting for topology to be initialized...");
-                Thread.sleep(5000);
-            }
-            while (!mockServiceStarted()) {
+
+            // checking whether mock iaas component is ready. If it is ready, all the components are activated.
+            mockIaasApiClient = new MockIaasApiClient(stratosTestServerManager.getWebAppURL() + MOCK_IAAS_API_EP);
+            while (!StratosServerExtension.mockIaasApiClient.isMockIaaSReady()) {
                 log.info("Waiting for mock service to be initialized...");
                 Thread.sleep(1000);
```

Comment:
server is started from StratosTestServerManager#startServerUsingCarbonHome(). There we are waiting until the server is started. So I didn't "Waiting for mock service to be initialized..." message continuously. Waiting 50s before entering loop will be a significant delay, wdyt?